src/main/java/com/icthh/xm/uaa/security/oauth2/idp/jwk/EllipticCurveJwkDefinition.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.icthh.xm.uaa.security.oauth2.idp.jwk;
+
+/**
+ * This class copied from {@link org.springframework.security.oauth2.provider.token.store.jwk.EllipticCurveJwkDefinition}.
+ * and couldn't be imported cause of package private access.
+ * A JSON Web Key (JWK) representation of an Elliptic Curve key.
+ *
+ * @author Michael Duergner <michael@sprucehill.io>
+ * @since 2.3
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7517">JSON Web Key (JWK)</a>
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7518#page-28">JSON Web Algorithms (JWA)</a>
+ */
+public final class EllipticCurveJwkDefinition extends JwkDefinition {
```

Comment:
why we copy paste source code on spring-security-oauth2 ?

how about migrations, bug fixes etc?

Reply:
We copy source code from the spring-security-oauth2 package only for one reason - we have demands to customize the behavior of some methods, that involved in the token validation chain and cause of original spring-security-oauth2 has package-private access class/methods within it is impossible to customize them.